pkg/controller/integrationcontext/initialize.go

```diff
@@ -45,19 +47,24 @@ func (action *initializeAction) CanHandle(ictx *v1alpha1.IntegrationContext) boo
 
 func (action *initializeAction) Handle(ctx context.Context, ictx *v1alpha1.IntegrationContext) error {
 	// The integration platform needs to be initialized before starting to create contexts
-	if _, err := platform.GetCurrentPlatform(ctx, action.client, ictx.Namespace); err != nil {
+	p, err := platform.GetCurrentPlatform(ctx, action.client, ictx.Namespace)
+	if err != nil {
```

Comment:
wondering if we should chek `p` for `nil` to determine if a platform is there and use `err` to effectively signalling an error (and the context should be moved to the error state too)

Reply:
Makes sense. A good use case for #594 too.